### DISCUSSION TITLE
Filter alert rules by health

### DISCUSSION THREAD

**Discussion by adeverteuil** [grafana/grafana#57272] [Alerting]
In the alert rules page, I can filter rules by state (firing, normal, pending). I would also like the ability to filter by health (err or ok).

![Screenshot 2022-10-19 08-06-00 alert rules blur](https://user-images.githubusercontent.com/3270755/196687904-7b0b43b6-1c74-4341-b84c-9bebea77021f.png)

A rule health is "err" if the query expression has a syntax error, or has other problems that prevent the query engine to execute the rule.

I can find rules in error state by scrolling through the list. I can get the error message by a mouseover on the error.

![Screenshot 2022-10-19 08-14-42 error detail](https://user-images.githubusercontent.com/3270755/196687946-bdc1d1be-803c-44c7-9aed-b7d98e40528c.png)

As a workaround, I use the API to get the rules from Prometheus and Loki, and I use `jq` to filter the alert rules with an error state.

```
curl -u $MIMIR_TENANT_ID:$MIMIR_API_KEY \
$MIMIR_ADDRESS/prometheus/api/v1/rules \
| jq ".data.groups[].rules[]|select(.health == \"err\")"
```

**Comment by gtirloni** [comment]
This exists in Grafana 10
